File: pset1/mario/mario.c

#include <stdio.h>
#include <cs50.h>

int get_height(void);
void make_pyramid(int height);

int main(void)

{
    make_pyramid(get_height());
}

int get_height(void)

{

    int height;

    do
    {

        height = get_int("Height: ");


    }
    while (height < 1 || height > 8);

    return height;

}

void make_pyramid(int height)

{

    for (int i = 1; i <= height; i++)
    {

        for (int j = 0; j <= i * 2; j++ )
        {

            if (j == 0)
            {
              for (int z = 0; z < height - i; z++)

              {
                  printf(" ");
              }


            }

            if (j == (i*2)/2)

            {
                printf(" ");
                printf(" ");

            }

            else

            {
                printf("#");
            }
        }

        printf("\n");

    }


}

File: pset6/mario/more/mario.py
from cs50 import get_int

def create_pyramid(height):

    for i in range(height):
        n = height - (i + 1)
        print(" " * n, end = '')

        for j in range(((i + 1) * 2) + 1):
            if j == i + 1:
                print('  ', end = '')
            else:
                print("#", end = '')
        print()


height = get_int("Height: ")

while height < 1 or height > 8:
    height = get_int("Height: ")

create_pyramid(height)